edb/pgsql/compiler/dml.py

```diff
@@ -1488,6 +1473,23 @@ def process_link_values(
                     row_query, ir_stmt.subject.path_id,
                     env=input_rel_ctx.env),)
 
+            # Look for a shape that might provide link property values
```

Comment:
Let's not forget to remove this hack once we implement linkprop stickiness on derived types.

Reply:
I'm not sure that linkprop stickiness will help us with that?

I think we could get rid of this part of the hack by adding something like `nested_shape_source: Optional[Set]`, to Set, that `_compile_view_shapes_in_set` populates? So that the rummaging around only happens in one place instead of two?

(I think maybe we could also get rid of the rummaging around with a viewgen refactor, but I'm not sure)